### PR TITLE
Replace layer choice with selected module after applied fixed architecture

### PR DESCRIPTION
**Pull request by ultmaster**
This should solve the problem #2401 is trying to solve.

### CODE REVIEW

File: src/sdk/pynni/nni/nas/pytorch/fixed.py [hunk 1]
```diff
@@ -73,6 +77,36 @@ def sample_final(self):
...
+        Parameters
+        ----------
+        module : nn.Module
+            Module to be processed.
+        prefix : str
+            Module name under global namespace.
+        """
+        if module is None:
+            module = self.model
+        for name, mutable in module.named_children():
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/nas/pytorch/fixed.py:95]: Why not use the util function to traverse here?

File: src/sdk/pynni/nni/nas/pytorch/fixed.py [hunk 2]
```diff
@@ -73,6 +77,36 @@ def sample_final(self):
...
+            global_name = (prefix + "." if prefix else "") + name
+            if isinstance(mutable, LayerChoice):
+                chosen = self._fixed_arc[mutable.key]
+                if sum(chosen) == 1 and max(chosen) == 1 and not mutable.return_mask:
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/pytorch/fixed.py:99]: what is the meaning of `not mutable.return_mask`?

Comment (ultmaster) [reply src/sdk/pynni/nni/nas/pytorch/fixed.py:99]: If true, layer choice will return the choice vector. If we replace the layer choice module with the selected module, choice vector will be no longer available, and users’ code will crash.

**Comment by QuanluZhang** [comment]
please update doc at proper place.

File: docs/en_US/NAS/NasGuide.md [hunk 3]
```diff
@@ -176,7 +176,7 @@ For example,
 }
 ```
 
-After applying, the model is then fixed and ready for final training. The model works as a single model, although it might contain more parameters than expected. This comes with pros and cons. The good side is, you can directly load the checkpoint dumped from supernet during the search phase and start retraining from there. However, this is also a model with redundant parameters and this may cause problems when trying to count the number of parameters in the model. For deeper reasons and possible workarounds, see [Trainers](./NasReference.md).
+After applying, the model is then fixed and ready for final training. The model works as a single model, and unused parameters and modules will be pruned.
```
Comment (QuanluZhang) [inline docs/en_US/NAS/NasGuide.md:179]: "will be" -> "are"

**Pull request merged**